passes/source2il.nim

```diff
@@ -482,6 +482,50 @@ proc exprToIL(c; t: InTree, n: NodeIndex, bu, stmts): SemType =
       c.error("undeclared identifier: " & t.getString(n))
       bu.add Node(kind: IntVal)
       result = prim(tkError)
+  of SourceKind.If:
+    case t.len(n)
+    of 2: # condition and body
+      bu.subTree If:
+        let (p, b) = t.pair(n) # predicate and body
+        let cond = exprToIL(c, t, p, bu, stmts) # condition
+        if cond.kind != tkBool:
+          c.error("`If` condition must be a boolean expression")
+        let body = exprToIL(c, t, b, bu, stmts) # body
+        if body.kind notin {tkUnit, tkVoid}:
+          c.error("non-exhaustive `If` must be of type unit or void")
+          result = errorType()
+        else:
+          result = body
```

Comment:
OK, if I understand correctly an `if-then` should always be `tkUnit`, unless it's an error `tkError`.

The only way it could be `tkVoid` is if we had some sort of analysis that confirmed the condition always evaluated to `true` and the body was `tkVoid`. But we're not going to bother with this analysis right now, possibly ever.